lib/ansible/plugins/terminal/routeros.py

```diff
@@ -0,0 +1,72 @@
+#
+# (c) 2016 Red Hat Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import re
+
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils._text import to_text, to_bytes
+from ansible.plugins.terminal import TerminalBase
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class TerminalModule(TerminalBase):
+
+    ansi_re = [
+        # check ECMA-48 Section 5.4 (Control Sequences)
+        re.compile(br'(\x1b\[\?1h\x1b=)'),
+        re.compile(br'((?:\x9b|\x1b\x5b)[\x30-\x3f]*[\x20-\x2f]*[\x40-\x7e])'),
+        re.compile(br'\x08.')
+    ]
+
+    terminal_initial_prompt = [
+        br'\x1bZ',
+    ]
+
+    terminal_initial_answer = b'\x1b/Z'
+
+    terminal_stdout_re = [
+        re.compile(br"\x1b<"),
+        re.compile(br"\[\w+\@[\w\-\.]+\] ?> ?$"),
+        re.compile(br"Please press \"Enter\" to continue!"),
+        re.compile(br"Do you want to see the software license\? \[Y\/n\]: ?"),
+    ]
+
+    terminal_stderr_re = [
+        re.compile(br"\nbad command name"),
+        re.compile(br"\nno such item"),
+        re.compile(br"\ninvalid value for"),
+    ]
+
+    def on_open_shell(self):
+        prompt = self._get_prompt()
+        try:
+            if prompt.strip().endswith(br"\x1b<") or prompt.strip().endswith(b':'):
```

Comment:
I think you could simply this to `if prompt.strip().endswith(b':'):` here (at least, that works for me), since the first condition is probably never matched. `br"\x1b<"` is the sequence `\`, `x`, `1`, `b`, `<`, while you probably meant `b"\x1b<"`, which is `ESC`, `<`. Anyway, when I change `br"\x1b<"` to `b"\x1b<"`, it doesn't work anymore.

Reply:
In case you want to know, the error I get when changing `br"\x1b<"` to `b"\x1b<"` is:
```
The full traceback is:
Traceback (most recent call last):
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 243, in exec_command
    cmd = json.loads(to_text(cmd, errors='surrogate_or_strict'))
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 2 (char 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/paramiko/channel.py", line 683, in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "/usr/lib/python3.6/site-packages/paramiko/buffered_pipe.py", line 160, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 437, in send
    response = self.receive(command, prompt, answer, newline, prompt_retry_check)
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 400, in receive
    data = self._ssh_shell.recv(256)
  File "/usr/lib/python3.6/site-packages/paramiko/channel.py", line 685, in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../ansible/lib/ansible/plugins/terminal/routeros.py", line 68, in on_open_shell
    self._exec_cli_command(b' ')
  File ".../ansible/lib/ansible/plugins/terminal/__init__.py", line 73, in _exec_cli_command
    return self._connection.exec_command(cmd)
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 253, in exec_command
    return self.send(command=cmd)
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 441, in send
    raise AnsibleConnectionFailure("timeout trying to send command: %s" % command.strip())
ansible.errors.AnsibleConnectionFailure: timeout trying to send command: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../ansible/bin/ansible-connection", line 87, in start
    self.connection._connect()
  File ".../ansible/lib/ansible/plugins/connection/network_cli.py", line 328, in _connect
    self._terminal.on_open_shell()
  File ".../ansible/lib/ansible/plugins/terminal/routeros.py", line 72, in on_open_shell
    raise AnsibleConnectionFailure('unable to bypass license prompt')
ansible.errors.AnsibleConnectionFailure: unable to bypass license prompt

fatal: [router]: FAILED! => {
    "msg": "unable to bypass license prompt"
}
```